File: 58_length_if_last_word.py
#Start from end skip the initial space characters and start counting
# as soon as you reach a non-space character till you reach the next space charater.
# TC: O(N) SC: O(1)
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        # s = s.strip()
        # words = s.split(" ")
        # return len(words[-1])
        cur_index  = len(s)
        length = 0
        while cur_index > 0:
            cur_index -= 1
            if length > 0 and s[cur_index] == " ":
                return length
            if s[cur_index] != " ":
                length += 1
        return length



        

File: 58_length_if_last_word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        s=" "+s;
        int c=0, i=s.size()-1;
        for( ; i>-1&&!isalpha(s[i]); i--);
        for( ; i>-1&&s[i]!=' '; i--)
            c++;
        return c;
    }
};